### DISCUSSION TITLE
Setting Date of 1st release and date of next BASF-Evonik Bofire Workshop-Hackathon (to be hosted in LU) and frequency

### DISCUSSION THREAD

**Discussion by rosonaeldred** [experimental-design/bofire#112] [General]
Hey @jduerholt  @jkeupp @evo-bschiller @jkleinekorte ! 
We loosely discussed future meetings and their frequency during the hackathon. 

I suggested roughly once a quarter. Perhaps: set a release date and meet after to plan the next milestone and get started. 
@R-M-Lee discussed hosting a Hackathon in LU and has permission (and some funding for lunch for our guests). 

So:  Questions: 

1. When should the first release be? 
2. What rough timing would work on your end for a next workshop? (Would you be able to travel to us? )
3. What do you see as plausible/good frequency for these events?

@R-M-Lee  @bertiqwerty @dlinzner-bcs @WaStCo and myself thought maybe April (working around Easter...) ?

**Comment by jkeupp** [comment]
Hi, and thanks for reaching out!
I'd be fine with quarterly, and would be happy to join in in LU. Around easter would work for me. Regarding the more distant future, there's also an option to meet up here in IH.  

Regarding the first release: I'm not sure as to what the status of your ports are, and how many "known bugs" are yet to be fixed.
 I've planned to "finalize" #105, the PR I was part of by the end of this weekend for someone to review :P.
For then I'd be happy to do some more deep-dive into the actual inner workings, as opposed to creating usage examples (I'll continue to add examples, however. I'll have to explore w.r.t. some particular use-cases anyways). There is an issue already which i'll try to work on (it's assigned to me already and should not be too much of an issue) as a way to get this started.

**Comment by rosonaeldred** [comment]
![image](https://user-images.githubusercontent.com/36136386/220947729-2cd23116-1411-40fa-bbb7-55a4da766306.png)
Go vote! 
Dates are the point, times are flexible.

**Comment by sgbaird** [comment]
Excuse the nosy question. Is the hackathon mentioned for continuing the development of BoFire?

**Reply by R-M-Lee** [reply]
Nosy questions are quite welcome :) 

Yes, we already met up in April and I expect we will have another hackathon-type meeting at some later point in the year (although not yet agreed/organized). Are you interested in using/contributing/joining?

**Reply by sgbaird** [reply]
Hi @R-M-Lee, I'm interested in using and contributing to BoFire. Joining would depend on the location / whether there's a virtual option :)

@rosonaeldred, that was the sense I was getting, but I wanted to make sure. Thank you for clarifying!

**Reply by R-M-Lee** [reply]
awesome! Well just take a look and see if there's anything you want to improve or implement. We will keep you in mind if and when a future meeting is organized. A virtual part shouldn't be a problem

**Reply by sgbaird** [reply]
I want to make teaching-focused template generators for various optimization platforms like Ax, BoFire, [Olympus](https://github.com/aspuru-guzik-group/olympus), and [Chaos](https://github.com/schwallergroup/chaos). The idea isn't so much that you specify "I want 5 continuous variables in my search space": the idea is more that you get to mix and max advanced optimization topics. For example:

> Provide a script that shows how to perform batch multi-objective Bayesian optimization initialized with existing data

I envision the interface to look something like the PyTorch installation docs. See https://github.com/facebook/Ax/issues/1479 and https://discuss.pytorch.org/t/what-is-the-pytorch-installation-interactive-grid-called-and-how-do-i-make-my-own/173835/4

**Reply by jduerholt** [reply]
Hi @sgbaird,

sounds like a very nice idea, do you want to realize it in a next hackathon or already before? If before, do you need any help?

Best,

Johannes

**Reply by sgbaird** [reply]
@jduerholt thanks! Ideally, I'd like to have something to present by August 22 for [Accelerate 2023](https://www.accelerate23.ca/). I would love to get some help with it, especially with [the high-level design](https://github.com/sgbaird/honegumi/discussions/2)! I'm getting things rolling at https://github.com/sgbaird/honegumi. For starters, do you mind taking a look at https://colab.research.google.com/github/sgbaird/honegumi/blob/main/notebooks/1.0-sgb-gentle-introduction-jinja.ipynb and letting me know your thoughts?

**Comment by rosonaeldred** [comment]
The question might be actually  about the term “hackathon”.
For ‘normal people’, I’d call it more like a contribution workshop or so.


From: Robert Lee ***@***.***>
Sent: Friday, June 23, 2023 11:49 AM
To: experimental-design/bofire ***@***.***>
Cc: Rosona Eldred ***@***.***>; Author ***@***.***>
Subject: [EXT] Re: [experimental-design/bofire] Setting Date of 1st release and date of next BASF-Evonik Bofire Workshop-Hackathon (to be hosted in LU) and frequency (Discussion #112)


Nosy questions are quite welcome :)

Yes, we already met up in April and I expect we will have another hackathon-type meeting at some later point in the year (although not yet agreed/organized). Are you interested in using/contributing/joining?
